src/v/net/server.cc

```diff
@@ -264,46 +267,46 @@ server::accept_finish(ss::sstring name, ss::future<ss::accept_result> f_cs_sa) {
       this->name(),
       ar.remote_address,
       name);
-    if (_as.abort_requested() || _conn_gate.is_closed()) {
-        conn->shutdown_input();
-        co_await conn->shutdown();
-        throw ss::gate_closed_exception();
-    }
+
+    _as.check();
```

Comment:
>  `_as.check()` call in`server::accept_finish` is just an optimization after this change and not required for correctness.

this can still fire in some cases, right?

* shutdown_input() -> _as.request_abort_ex(ssx::shutdown_requested_exception{});
* \<scheduling point\>
* this resumes and _as.check() throws..

we should probably let wait_for_shutdown() do an orderly shutdown rather than letting _as.check() throw here? (or maybe I'm misreading the code).

Reply:
Is this throw problematic? I'm using it as a shortcut for `if (_as.abort_requested()) return;`. The accept routine is inside `spawn_with_gate` which wraps it further with `ignore_shutdown_exceptions`.

Reply:
Explained it here https://github.com/redpanda-data/redpanda/pull/13739/files#r1339951238

Reply:
its not problematic, was just curious if we need it, ok to keep it as-is.